mars/web/apihandlers.py

```diff
@@ -147,18 +148,12 @@ def _wait_fun():
         except GraphNotExists:
             raise web.HTTPError(404, 'Graph not exists')
 
-        if state == GraphState.RUNNING:
-            self.write(json.dumps(dict(state='running')))
-        elif state == GraphState.SUCCEEDED:
-            self.write(json.dumps(dict(state='success')))
-        elif state == GraphState.FAILED:
-            self.write(json.dumps(dict(state='failed')))
-        elif state == GraphState.CANCELLED:
-            self.write(json.dumps(dict(state='cancelled')))
-        elif state == GraphState.CANCELLING:
-            self.write(json.dumps(dict(state='cancelling')))
-        elif state == GraphState.PREPARING:
-            self.write(json.dumps(dict(state='preparing')))
+        if state == GraphState.FAILED:
+            exc = self.web_api.get_graph_traceback(session_id, graph_key)
+            if exc is not None:
+                self.write(json.dumps(dict(state=state.value,
+                                           traceback=''.join(traceback.format_exception(*exc)))))
```

Comment:
I wonder if you can put pickled traceback data here if possible to unify the behavior of actor clients and web clients? String traceback can be a backup.

Reply:
Fixed. Now the web session client return both the pickled `exc_info` and the textual traceback.